test/e2e/e2e_suite_test.go

```diff
@@ -218,6 +218,7 @@ func setupBootstrapCluster(config *clusterctl.E2EConfig, scheme *runtime.Scheme,
 			RequiresDockerSock: config.HasDockerProvider(),
 			Images:             config.Images,
 			IPFamily:           config.GetVariable(IPFamily),
+			LogFolder:          filepath.Join(artifactFolder, "kind"),
```

Comment:
Ideally, this change should be implemented by everyone using the E2E test framework.
An option to give people room to this change overtime is to log only if the log folder is defined.

Reply:
Right now the field is not required but if the cluster creation fails it would log into folder "", right?

I would prefer one of the following options:
1. enforce that the log folder is set (via an `Expect` at the beginning of `CreateKindBootstrapClusterAndLoadImages`)
1. only export logs if the LogFolder is set

Both options are fine for me, I would prefer 1.. Afaik our API compatibility guarantees for framework allow this and as log folder is used everywhere it shouldn't be a problem for users to also set it here. Option 2. would probably mean that apart from us probably almost nobody benefits of this improvement.

(Would be good to mention this in the migration guide if we implement Option 1 though)

Reply:
We are now exporting logs only if the LogFolder is set; this is documented in the migration guide (even if it is not mandatory)